### ISSUE TITLE
Corrupted exports

### ISSUE THREAD

**Issue by ryandesign** [AppleCommander/AppleCommander#134]
Files I export from AppleCommander 1.9.0 on macOS 12.7.1 are corrupted. For example, boot an Apple DOS 3.3 disk image in an emulator and save the ROM to the disk image with:

```
BSAVE ROM,A$D000,L$3000
```

Open that disk image in AppleCommander and export the ROM file. Open the exported ROM file in a hex editor. On my system, there were two 256-byte regions of the file that were filled with zeroes when of course the real ROM does not have those gaps, and if I open the disk image itself into the hex editor those gaps are not present.

**Comment by ryandesign** [comment]
The same problem is evident if I View the file in AppleCommander rather than export it.

**Comment by ryandesign** [comment]
I now suspect this is a bug in the way that the Clock Signal emulator is writing to disk images, not the way that AppleCommander is reading them. I've done some more tests with a 16K plain text file, and the file does not show gaps when it was written by AppleCommander or Open Emulator or Virtual ][, only when it was written by Clock Signal. This and my previous test with the ROM file was on a disk image with other files on it. I attempted to reproduce the ROM-writing test on a blank disk image created by AppleCommander, and Open Emulator and Virtual ][ created identical disks; meanwhile Clock Signal now expresses `I/O ERROR` in response to the `BSAVE ROM,A$D000,L$3000` command.

Before realizing that, I looked at the AppleCommander file reading code, and it looks like DOS maintains a list of tracks and sectors that contain the file's data. Since a sector is 256 bytes, which matches the size of each gap, I wonder if Clock Signal has an off-by-one error somewhere that causes it to skip writing a sector sometimes, or to write the sector but not include it in the track/sector map.
